framework/src/userobject/ActivateElementUserObject.C

```diff
@@ -0,0 +1,535 @@
+
+ActivateElementUserObject::ActivateElementUserObject(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _active_subdomain_id(
+        declareRestartableData<int>("active_subdomain_id", getParam<int>("active_subdomain_id"))),
+    _inactive_subdomain_id(declareRestartableData<int>("inactive_subdomain_id",
+                                                       getParam<int>("inactive_subdomain_id"))),
+    _expand_boundary_name(getParam<std::vector<BoundaryName>>("expand_boundary_name")),
+    _variable_activation(
+        declareRestartableData<bool>("variable_activation", getParam<bool>("variable_activation"))),
+    _function_x(isParamValid("function_x") ? &getFunction("function_x") : nullptr),
+    _function_y(isParamValid("function_y") ? &getFunction("function_y") : nullptr),
+    _function_z(isParamValid("function_z") ? &getFunction("function_z") : nullptr),
+    _activate_distance(
+        declareRestartableData<Real>("activate_distance", getParam<Real>("activate_distance"))),
+    _coupled_var(isParamValid("coupled_var") ? &coupledValue("coupled_var") : nullptr),
+    _activate_value(
+        declareRestartableData<Real>("activate_value", getParam<Real>("activate_value")))
+{
+  if (_variable_activation && _coupled_var == nullptr)
+    paramError("variable_activation",
+               "Need to specify a 'coupled_var' if variable_activation = true");
+
+  if ((!_variable_activation) && _coupled_var != nullptr)
+    mooseWarning("Not using variable activation, so the 'coupled_var' is ignored");
+
+  setNewBoundayName();
+}
+
+void
+ActivateElementUserObject::setNewBoundayName()
+{
+  // add the new boundary and get its boundary id
+  _boundary_ids = _mesh.getBoundaryIDs(_expand_boundary_name, true);
+  _mesh.setBoundaryName(_boundary_ids[0], _expand_boundary_name[0]);
+  _mesh.getMesh().get_boundary_info().sideset_name(_boundary_ids[0]) = _expand_boundary_name[0];
+  _mesh.getMesh().get_boundary_info().nodeset_name(_boundary_ids[0]) = _expand_boundary_name[0];
+
+  auto displaced_problem = _fe_problem.getDisplacedProblem();
+  if (displaced_problem)
+  {
+    _disp_boundary_ids = displaced_problem->mesh().getBoundaryIDs(_expand_boundary_name, true);
+    displaced_problem->mesh().setBoundaryName(_disp_boundary_ids[0], _expand_boundary_name[0]);
+    displaced_problem->mesh().getMesh().get_boundary_info().sideset_name(_disp_boundary_ids[0]) =
+        _expand_boundary_name[0];
+    displaced_problem->mesh().getMesh().get_boundary_info().nodeset_name(_disp_boundary_ids[0]) =
+        _expand_boundary_name[0];
+  }
+}
+
+void
+ActivateElementUserObject::execute()
+{
+  bool activate_by_var = false;
+  bool activate_by_path = false;
+  /*
+    Check if current element is activated
+  */
+  if (!_variable_activation)
+  {
+    // activate center (assume position of the activate center is only time dependent)
+    Real x_t = _function_x->value(_t, _q_point[0]);
+    Real y_t = _function_y->value(_t, _q_point[0]);
```

Comment:
So here the path function is only dependent on the time. So I pass a dummy `Point` here to the `Function::value()`. I agree it is confusing. Any ideas to improve it here?

Reply:
Actually use a dummy point

```
const static Point dummy;
Real x_t = _function_x->value(_t, dummy);
```

otherwise it looks like this has to be the QP

Reply:
`const static` means this dummy doesn't have to be rebuilt on the stack each time you enter the function, and it indicates to the reader that it's value does not change from `(0,0,0)`.

Reply:
> `const static` means this dummy doesn't have to be rebuilt on the stack each time you enter the function, and it indicates to the reader that it's value does not change from `(0,0,0)`.

Thanks! Looks like the best way to go